ethereum/core/src/jmh/java/org/hyperledger/besu/ethereum/vm/operations/DivOperationBenchmark.java

```diff
@@ -16,12 +16,115 @@
 
 import org.hyperledger.besu.evm.frame.MessageFrame;
 import org.hyperledger.besu.evm.operation.DivOperation;
+import org.hyperledger.besu.evm.operation.DivOperationOptimized;
 import org.hyperledger.besu.evm.operation.Operation;
 
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Random;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+
 public class DivOperationBenchmark extends BinaryOperationBenchmark {
 
+  private static final Random RANDOM = new Random();
+
+  // public because of jmh code generator
+  public enum MODE {
+    ZERO_QUOTIENT,
+    ONE_QUOTIENT,
+    LARGER_QUOTIENT,
+    UNIFORM_OPERATORS,
+    INT_OPERATORS,
+    LONG_OPERATORS,
+    SMALL_QUOTIENT,
+    BIG_OPERATORS,
+    FULL_RANDOM
+  }
+
+  public enum ALGO {
+    NON_OPTIMIZED,
+    OPTIMIZED
+  }
+
+  @Param private MODE mode;
+
+  @Param protected ALGO algorithm;
+
+  @Setup
+  @Override
+  public void setUp() {
+    frame = BenchmarkHelper.createMessageCallFrame();
+
+    switch (mode) {
+      case ZERO_QUOTIENT:
+        aPool = new Bytes[SAMPLE_SIZE];
+        bPool = new Bytes[SAMPLE_SIZE];
+        Arrays.fill(aPool, Bytes.EMPTY);
+        BenchmarkHelper.fillPool(bPool, () -> 1 + RANDOM.nextInt(32));
+        break;
+      case ONE_QUOTIENT:
+        aPool = new Bytes[SAMPLE_SIZE];
+        BenchmarkHelper.fillPool(aPool, () -> 1 + RANDOM.nextInt(32));
+        bPool = aPool;
+        break;
+      case LARGER_QUOTIENT:
+        fillPools(
+            () -> 22 + RANDOM.nextInt(10),
+            () -> 1 + RANDOM.nextInt(10),
+            byteArray -> new BigInteger(1, byteArray),
+            (__, ___) -> true);
+        break;
+      case SMALL_QUOTIENT:
+        fillPools(
+            () -> 24 + RANDOM.nextInt(9),
+            () -> 24 + RANDOM.nextInt(9),
+            byteArray -> new BigInteger(1, byteArray),
+            (a, b) -> a.compareTo(b) > 0);
+        break;
+      case UNIFORM_OPERATORS:
+        fillPools(
+            () -> 1 + RANDOM.nextInt(32),
+            () -> 1 + RANDOM.nextInt(32),
+            byteArray -> new BigInteger(1, byteArray),
+            (a, b) -> a.compareTo(b) > 0);
+        break;
+      case INT_OPERATORS:
+        fillPools(
+            () -> 5 + RANDOM.nextInt(4),
+            () -> 1 + RANDOM.nextInt(4),
+            byteArray -> new BigInteger(1, byteArray),
+            (__, ___) -> true);
```

Comment:
I don't a case where true is needed, why not change the order of the two first parameters and set the last parameter to false ?